worksheet_D/oxo.py

```diff
@@ -0,0 +1,131 @@
+class OxoBoard:
+
+    def __init__(self):
+        """Define board coordinates and set values to 0"""
+        self.oxoboard = {(0, 0): 0, (1, 0): 0, (2, 0): 0,
+                         (0, 1): 0, (1, 1): 0, (2, 1): 0,
+                         (0, 2): 0, (1, 2): 0, (2, 2): 0}
+
+
+    def get_square(self, x, y):
+        """ Return 0, 1 or 2 depending on the contents of the specified square. """
+        self.x = x
+        self.y = y
+        return self.oxoboard[x, y]
+
+
+    def set_square(self, x, y, mark):
+        """ If the specified square is currently empty (0), fill it with mark and return True.
+            If the square is not empty, leave it as-is and return False. """
+        if self.oxoboard[x, y] == 0:
+            self.oxoboard[x, y] = mark
+            return True
+        else:
+            return False
+
+
+    def is_board_full(self):
+        """ If there are still empty squares on the board, return False.
+            If there are no empty squares, return True. """
+        squares = 9
+        for i in xrange(0, 3):
+            for j in xrange(0, 3):
+                if self.oxoboard[i, j] > 0:
+                    squares -= 1
+                if squares == 0:
+                    return True
+        return False
+
+
+    def get_winner(self):
+        """ If a player has three in a row, return 1 or 2 depending on which player.
+            Otherwise, return 0. """
+        current_player = 1
+        # Switch players
+        if current_player == 1:
+            current_player = 2
+        else:
+            current_player = 1
+        if self.oxoboard[0,0] != 0 and self.oxoboard[0,0] == self.oxoboard[1,0] and self.oxoboard[0,0] == self.oxoboard[2,0]:
+            return current_player
```

Comment:
This is not correct -- you should instead return the player number for whoever owns this line. For example you could return `self.oxoboard[0,0]` in this case.